src/pages/Transaction/index.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { Switch, Route, useRouteMatch } from 'react-router-dom';
+
+import MainContainer from 'parts/MainContainer';
+import { URL_PARAMETERS } from 'utils/constants/links';
+
+const IssueTransaction = React.lazy(() => import(/* webpackChunkName: 'issue-transaction' */ './IssueTransaction'));
+const RedeemTransaction = React.lazy(() => import(/* webpackChunkName: 'redeem-transaction' */ './RedeemTransaction'));
+const ReplaceTransaction = React.lazy(
+  () => import(/* webpackChunkName: 'replace-transaction' */ './ReplaceTransaction')
+);
+
+enum TransactionType {
+  Issue = 'issue',
+  Redeem = 'redeem',
+  Replace = 'replace'
+}
+
+const Transaction = (): JSX.Element => {
+  const { path } = useRouteMatch();
+
+  return (
+    <MainContainer>
+      <Switch>
+        <Route path={`${path}/${TransactionType.Issue}/:${URL_PARAMETERS.TRANSACTION_HASH}`}>
+          <IssueTransaction />
+        </Route>
+        <Route path={`${path}/${TransactionType.Redeem}/:${URL_PARAMETERS.TRANSACTION_HASH}`}>
+          <RedeemTransaction />
+        </Route>
+        <Route path={`${path}/${TransactionType.Replace}/:${URL_PARAMETERS.TRANSACTION_HASH}`}>
+          <ReplaceTransaction />
+        </Route>
+      </Switch>
```

Comment:
Nested routes under `/transaction`

Reply:
I think it could be better if we didn't had both `/transaction` and `/transactions` paths. Maybe if this component was merged with `pages/Transactions` component we could have unified endpoint `transactions/{transactionType}/{transactionHash}` ( so using plural for collection also according to [REST resource naming conventions](https://restfulapi.net/resource-naming/)). What are your thoughts on this? cc @tomjeatt

Reply:
That's a good idea in general and it's also one of the best practices in naming URIs. This case is a little different. I'm sorry for confusing. The `/transaction` endpoint is quite different from `/transactions` one from a feature perspective. It's for debugging and looking into a single transaction (issue/redeem/replace and so on) mostly for developers while `/transactions` is for listing issue/redeem transactions for users and it should not get involved with `/transactions`. I could name it better e.g. `tx-details` or something but we can improve it progressively. The point is that the two routes should not be related to each other at least for now. Maybe later on if we want to combine them together, I think we should follow this practice. Does it sound reasonable?

Reply:
Thank you for providing this information :) I wasn't aware that it is for debug purposes, in that case it makes total sense to do it like this :+1: